commit f73133e6ca71a218212456c6dcf1c00970727610
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sun Dec 18 01:32:57 2022 +0100

    disable threaded linking for kernel
    
    This drastically reduces memory usage in some environments.

diff --git a/Makefile b/Makefile
index 997b677..ab7ba02 100644
--- a/Makefile
+++ b/Makefile
@@ -538,7 +538,7 @@ LDFLAGS_MODULE  =
 CFLAGS_KERNEL	=
 RUSTFLAGS_KERNEL =
 AFLAGS_KERNEL	=
-export LDFLAGS_vmlinux =
+export LDFLAGS_vmlinux = --threads=1
 
 # Use USERINCLUDE when you must reference the UAPI directories only.
 USERINCLUDE    := \
diff --git a/arch/riscv/Makefile b/arch/riscv/Makefile
index 0d13b59..4429578 100644
--- a/arch/riscv/Makefile
+++ b/arch/riscv/Makefile
@@ -14,6 +14,8 @@ ifeq ($(CONFIG_DYNAMIC_FTRACE),y)
 	CC_FLAGS_FTRACE := -fpatchable-function-entry=8
 endif
 
+LDFLAGS_vmlinux += --threads=1
+
 ifeq ($(CONFIG_CMODEL_MEDLOW),y)
 KBUILD_CFLAGS_MODULE += -mcmodel=medany
 endif
